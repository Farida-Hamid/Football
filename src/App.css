body {
  margin: auto;
  background: #f7fa98;
}
