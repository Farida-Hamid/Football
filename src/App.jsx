import { Routes, Route } from 'react-router-dom';
import Header from './components/header';
import Leagues from './components/leagues';
import { recieveLeagues } from './redux/teams';
import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Info from './components/info';
import 'bootstrap/dist/css/bootstrap.css';

function App() {
  const leagueList = useSelector(state => state);
  const dispatch = useDispatch();

  useEffect(() => async () => {
    await dispatch(recieveLeagues());
  }, []);

  return (
    <div>
      <Header />
      <main>
      {leagueList.length ? (
        <Routes>
          <Route path="/" element={<Leagues leagues={leagueList}/>} />
          <Route path="/info" element={<Info />} />
        </Routes>
        ) : ''}
      </main>
    </div>
  );
}

export default App;