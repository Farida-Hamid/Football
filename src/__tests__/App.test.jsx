import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import App from '../App';

it('Testing the root component, the application as a whole:', () => {
  const tree = TestRenderer.create(
    <Provider store={store}>
      <App />
    </Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
