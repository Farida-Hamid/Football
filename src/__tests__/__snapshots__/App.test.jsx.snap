// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the root component, the application as a whole: 1`] = `
<div>
  <nav>
    <div
      className="home"
    >
      <h1>
        Football Leagues
      </h1>
    </div>
    <div
      className="links"
    >
      links
    </div>
  </nav>
  <main />
</div>
`;
